BSseq_pipeline.py

```diff
@@ -256,10 +263,15 @@
         refGenes_bedfile  = config['general']['differential-methylation']['annotation']['refGenes_bedfile'],
         chrom_seqlengths  = os.path.join(DIR_mapped,"Refgen_"+ASSEMBLY+"_chromlengths.csv"),
         assembly    = ASSEMBLY,
-        treatment = lambda wc: str(wc.treatment).replace('vs', '_'),
+        treatment  = lambda wc: str(wc.treatment).replace('vs', '_'),
         qvalue     = float(config['general']['differential-methylation']['qvalue']),
         difference = float(config['general']['differential-methylation']['difference']),
-        webfetch    = config['general']['differential-methylation']['annotation']['webfetch']
+        webfetch    = config['general']['differential-methylation']['annotation']['webfetch'],
+        methylDiffBed           = lambda wc: DIR_diffmeth + str(wc.treatment).replace('vs', '_') + dedupe_tag(config["SAMPLES"][get_sampleids_from_treatment(wc.treatment[0])[0]]['Protocol']) + '_diffmeth.bed',
+        methylDiff_file         = lambda wc: DIR_diffmeth + str(wc.treatment).replace('vs', '_') + dedupe_tag(config["SAMPLES"][get_sampleids_from_treatment(wc.treatment[0])[0]]['Protocol']) + '_diffmeth.RDS',
+        methylDiff_hyper_file   = lambda wc: DIR_diffmeth + str(wc.treatment).replace('vs', '_') + dedupe_tag(config["SAMPLES"][get_sampleids_from_treatment(wc.treatment[0])[0]]['Protocol']) + '_diffmethhyper.RDS',
+        methylDiff_hypo_file    = lambda wc: DIR_diffmeth + str(wc.treatment).replace('vs', '_') + dedupe_tag(config["SAMPLES"][get_sampleids_from_treatment(wc.treatment[0])[0]]['Protocol']) + '_diffmethhypo.RDS',
+        methylDiff_nonsig_file  = lambda wc: DIR_diffmeth + str(wc.treatment).replace('vs', '_') + dedupe_tag(config["SAMPLES"][get_sampleids_from_treatment(wc.treatment[0])[0]]['Protocol']) + '_diffmethnonsig.RDS'
```

Comment:
This seems to duplicate some of the inputs.  Is this necessary?

It also seems to me that we could use a named procedure instead:

```
def makeDiffMethPath(suffix, wc):
    return DIR_diffmeth + str(wc.treatment).replace('vs', '_') + dedupe_tag(config["SAMPLES"][get_sampleids_from_treatment(wc.treatment[0])[0]]['Protocol']) + '_' + suffix
```

Then we could call it like this:

```
methylDiff_nonsig_file = makeDiffMethPath("diffmethnonsig.RDS", wc)
```

This seems better to me than to repeat the ugly lambda expression.

Reply:
Or even better: use "treatment" as the second argument instead of "wc".  Because then we could use that same procedure in the next hunk.